db/analysis.go

```diff
@@ -0,0 +1,124 @@
+package db
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/jackc/pgx/v4/pgxpool"
+	"github.com/likecoin/likecoin-chain-tx-indexer/logger"
+)
+
+func GetNftCount(conn *pgxpool.Conn, q QueryNftCountRequest) (count QueryCountResponse, err error) {
+	sql := `
+	SELECT COUNT(n.id)
+	FROM nft as n
+	JOIN nft_class AS c USING (class_id)
+	JOIN iscn as i ON c.parent_iscn_id_prefix = i.iscn_id_prefix
+	WHERE ($1 = true OR i.owner != n.owner)
+		AND ($2::text[] IS NULL OR n.owner != ALL($2))
+	`
+	ctx, cancel := GetTimeoutContext()
+	defer cancel()
+
+	err = conn.QueryRow(ctx, sql, q.IncludeOwner, q.IgnoreList).Scan(&count.Count)
+	if err != nil {
+		err = fmt.Errorf("get nft count failed: %w", err)
+		logger.L.Error(err, q)
+	}
+	return
+}
+
+func GetNftTradeStats(conn *pgxpool.Conn, q QueryNftTradeStatsRequest) (res QueryNftTradeStatsResponse, err error) {
+	payload := struct {
+		Type    string `json:"@type"`
+		Grantee string `json:"grantee"`
+	}{
+		"/cosmos.authz.v1beta1.MsgExec",
+		q.ApiAddress,
+	}
+	payloadJSON, err := json.Marshal(&payload)
+	if err != nil {
+		err = fmt.Errorf("marshal nft trade stats payload failed: %w", err)
+		logger.L.Error(err, q)
+		return
+	}
+
+	sql := `
+	SELECT count(*), sum((tx #>> '{"tx", "body", "messages", 0, "msgs", 0, "amount", 0, "amount"}')::bigint)
+	from txs
+	WHERE tx #> '{"tx", "body", "messages", 0}' @> $1
```

Comment:
This looks hacky as
(1) it only counts the trades done through us (by using the `authz` module)
(2) it only counts the first message in the tx, so if we changed backend code, this will suddenly break

Can we do this by query events from the nft / likernft modules?

Reply:
nft events doesn't include price info since they're different message. It's a hacky solution by @williamchong

Reply:
Can we use `MsgSend` from `nft` module sent from our wallet instead? @lancatlin 
not sure about if querying array would have huge performance impact for issue (2), MsgSend should always be in 2nd msg though

Reply:
Updated to filter by nft event. But still only counts volume through authz